torch/csrc/DynamicTypes.cpp

```diff
@@ -82,22 +88,38 @@ static PyTypeObject* getPyTypeObject(const at::Storage& storage)
   throw std::invalid_argument("unsupported Storage type");
 }
 
-THPDtype* getDtype(const at::Type& type) {
-  auto it = attype_to_dtype.find(&type);
-  if (it != attype_to_dtype.end()) {
-    return it->second;
+at::Type& getType(const THPDtype &dtype, const THPLayout& layout) {
+  at::Backend backend = get_backend(dtype.is_cuda, !layout.is_strided);
+  // use type_registry rather than context.getType() because getType throws exceptions.
+  auto baseType = at::globalContext().type_registry[static_cast<int>(backend)]
+                                                   [static_cast<int>(dtype.scalar_type)].get();
+  if (!baseType) {
+    std::ostringstream oss;
+    oss << "Error attempting to use dtype " << dtype.name << " with layout " << layout.name << ".";
+    if (dtype.is_cuda) {
```

Comment:
Looks like this condition isn't accurate. The contbuilds are showing "not compiled with CUDA enabled" error messages in configurations that are compiled with CUDA. (No fp16 sparse or something?)

Reply:
Good point; this bug probably already existed but was exposed because the test isn't properly skipping that case.  Maybe we can look through allDeclaredTypes so we don't have to special case the logic here.

Reply:
I added a cuda_enabled.h to torch/csrc/utils/cuda_enabled.h that just has a function to check this.